### ISSUE TITLE
Remove re-exposed IRF axis properties

### ISSUE THREAD

**Issue by cdeil** [gammapy/gammapy#1271]
I think we said that IRF axis properties from the NDData object shouldn't be re-exposed on the IRF object. Looks like we still do this currently in those cases:
```
$ ack 'return self\.data\.axis' gammapy
gammapy/scripts/cta_irf.py
286:        return self.data.axis('energy')

gammapy/irf/effective_area.py
79:        return self.data.axis('energy')

gammapy/irf/energy_dispersion.py
106:        return self.data.axis('e_reco')
111:        return self.data.axis('e_true')

gammapy/spectrum/core.py
80:        return self.data.axis('energy')
```
There's probably callers for that, so I'm splitting this out into a separate issue.

**Comment by cdeil** [comment]
Moving to v0.9, IRFs will need much more further thought / work.

Just a note: the thinking on this has reversed, the idea is now to make the NDDataArray an implementation detail on the IRF classes, and to expose what is needed to work with IRFs on the IRF class (re-exposing NDDataArray things for the IRFs that are implemented with this).

**Comment by adonath** [comment]
Solved via #3185.
